catalog/plugin/catalog-plugin-expirationdate/src/main/java/org/codice/ddf/catalog/plugin/expiration/ExpirationDatePlugin.java

```diff
@@ -0,0 +1,149 @@
+
+        LOGGER.debug("START Pre-Ingest Plugin: {}.process(CreateRequest createRequest)",
+                this.getClass().getName());
+
+        if (createRequest == null || createRequest.getMetacards() == null
+                || createRequest.getMetacards().isEmpty()) {
+            throw new PluginExecutionException("No metacards to validate or CreateRequest is null");
+        }
+
+        List<Metacard> metacards = createRequest.getMetacards();
+
+        for (Metacard metacard : metacards) {
+            updateExpirationDate(metacard);
+        }
+
+        LOGGER.debug("END Pre-Ingest Plugin: {}.process(CreateRequest createRequest)",
+                this.getClass().getName());
+        return createRequest;
+    }
+
+    @Override
+    public UpdateRequest process(UpdateRequest updateRequest)
+        throws PluginExecutionException, StopProcessingException {
+        return updateRequest;
+    }
+
+    @Override
+    public DeleteRequest process(DeleteRequest deleteRequest)
+        throws PluginExecutionException, StopProcessingException {
+        return deleteRequest;
+    }
+
+    public void setOffsetFromModifedDate(int days) {
+        LOGGER.debug("Setting offset from modified date to: {}", days);
+        this.offsetFromModifiedDate = days;
+    }
+
+    public void setOverwriteIfBlank(boolean overwriteIfBlank) {
+        LOGGER.debug("Setting overwrite if blank to: {}", overwriteIfBlank);
+        this.overwriteIfBlank = overwriteIfBlank;
+    }
+
+    public void setOverwriteIfExists(boolean overwriteIfExists) {
+        LOGGER.debug("Setting overwrite if exists to: {}", overwriteIfExists);
+        this.overwriteIfExists = overwriteIfExists;
+    }
+
+    /**
+     * Updates a metacard's expiration date.
+     *
+     * @param metacard
+     *            the metacard to update.
+     */
+    private void updateExpirationDate(Metacard metacard) {
+
+        Date currentExpirationDate = metacard.getExpirationDate();
+
+        if (currentExpirationDate == null && !overwriteIfBlank) {
+            // Don't overwrite empty expiration date if configuration disallows
+            LOGGER.debug(
+                    "The Expiration Date Pre-Ingest Plugin is not configured to overwrite 'empty' expiration dates. Not overwriting null expiration date for metacard ID [{}]. ",
+                    metacard.getId());
+            return;
+
+        } else if (currentExpirationDate != null && !overwriteIfExists) {
+            // Don't overwrite existing expiration date if configuration disallows
+            LOGGER.debug(
+                    "The Expiration Date Pre-Ingest Plugin is not configured to overwrite 'existing' expiration dates. Not overwriting the existing expiration date of {} for metacard ID [{}]. ",
+                    currentExpirationDate, metacard.getId());
+            return;
+        }
+
+        Date newExpirationDate = calculateNewExpirationDate(metacard.getModifiedDate());
```

Comment:
Modified date is supposed to be the modified date of the resource.  I _think_ we want to expire based on the metacard modified or created date (this is a new attribute in the taxonomy).  We should discuss this.